lib/compeon/rack_tools/pipes/parse_oauth2_authorization_code.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'compeon/rack_tools/http_errors'
+require 'compeon/rack_tools/pipes/log'
+
+module Compeon
+  module RackTools
+    module Pipes
+      PARSE_OAUTH2_AUTHORIZATION_CODE = lambda do |code:, **rest|
+        token = Compeon::AccessToken.parse(code)
```

Comment:
I think it's wrong what you do here. `AccessToken` is something else then an `OAUTH2_AUTHORIZATION_CODE`

Reply:
Oauth2 auth code is signed by auth service's private key. `AccessToken` utilizes Oauth2 public key. Why is it wrong?

Reply:
The tokens have a different structure.

Reply:
Yes, redirect uri instead of role. Alternative is to implement a parameter store lookup here and decode manually via JWT.

Reply:
call summary: We will inline the code for now and refactor it later to something like `Compeon::***::AuthToken`